zkvm/src/blockchain/state.rs

```diff
@@ -62,6 +63,29 @@ impl BlockchainState {
 
         Ok(())
     }
+
+    /// Executes a transaction, returning its tx ID and tx log.
+    pub fn execute_tx(
+        tx: &Tx,
+        bp_gens: &BulletproofGens,
+        block_version: u64,
+        timestamp_ms: u64,
+    ) -> Result<(TxID, TxLog), BlockchainError> {
+        if tx.header.mintime_ms > timestamp_ms || tx.header.maxtime_ms < timestamp_ms {
+            return Err(BlockchainError::BadTxTimestamp);
+        }
+
+        // Check that, for the current block version, this tx version is
+        // supported. For block versions higher than 1, we do not yet know
+        // what tx versions to support, so we accept all.
+        if block_version == 1 && block_version != tx.header.version {
```

Comment:
Per my previous comment, let's not compare tx version with block version. Ideally, they should be of different types and not comparable with each other ;-)

We want to say "if block version is known, the tx version must be known (per block version, so blkver=1 may allow only txver=1, but blkver=2 may allow txver={1,2})".

```suggestion
        if block_version == 1 && tx.header.version != 1 {
```

Reply:
got it! makes sense :)